orchagent/portsorch.cpp

```diff
@@ -2556,6 +2556,34 @@ void PortsOrch::doLagMemberTask(Consumer &consumer)
     }
 }
 
+void PortsOrch::doTask()
+{
+    auto tableOrder = {
+        APP_PORT_TABLE_NAME,
+        APP_LAG_TABLE_NAME,
+        APP_LAG_MEMBER_TABLE_NAME,
+        APP_VLAN_TABLE_NAME,
+        APP_VLAN_MEMBER_TABLE_NAME,
+    };
+
+    for (auto tableName: tableOrder)
+    {
+        auto consumer = getExecutor(tableName);
+        consumer->drain();
+    }
+
+    // drain remaining tables
+    for (auto& it: m_consumerMap)
+    {
+        auto tableName = it.first;
+        auto consumer = it.second.get();
+        if (find(tableOrder.begin(), tableOrder.end(), tableName) != tableOrder.end())
```

Comment:
>find(tableOrder.begin(), tableOrder.end(), tableName) != tableOrder.end() [](start = 12, length = 73)

The code shows if `tableName` is found in `tableOrder`, it will be `drain`ed.

I guess you wanted to `drain` ones not found in `tableOrder`.

Reply:
If my guess is correct, why your test pass?

---
In reply to: [353389068](https://github.com/Azure/sonic-swss/pull/1108#discussion_r353389068) [](ancestors = 353389068)

Reply:
You are correct, this is a bug. I'm fixing it.
Test passes because tableOrder contains all tables which PortsOrch uses, so no more tables had to be drained.